### ISSUE TITLE
Catching errors is hard

### ISSUE THREAD

**Issue by markknol** [ddj231/Handel#5]
I tried to catch the errors in my tiny editor to have some more direct feedback than using devtools, so I tried a few things but for some reason its hard to get the errors, would be nice if that could be solved.
```
<script src="https://unpkg.com/handel-pl"></script>
<textarea id=music style="height:50%">
start
	save c = C4, E4, G4 for 1b
	save f = F4, A4, C5 for 2b
	save g = G4, B4, D5 for 2b

	oops

	play c
	play f
	play g
finish
</textarea>
<textarea id=errors style="color:red;height:25%"></textarea>
<script>
function play() {
	errors.value = '';
	try {
		Handel.RunHandel(music.value); // also not promise, so not .catch
	} catch(e) {
		errors.value = `${e}`;  // nope
	}
}
window.onerror = e => errors.value = `${e}`; // nope
</script>
<button onclick="play()">Play</button>
<style>
body { background: #111; color: #ccc; }
textarea { width:100%;background: #333; color:#eee; padding:20px; font-size:17px; }
</style>
```

**Comment by ddj231** [comment]
@markknol thanks for submitting this issue. I've noticed the problem as well. The errors aren't being caught and rethrown, so they do not currently bubble up. This is something I can look to address.

**Comment by ddj231** [comment]
Hi @markknol, this now addressed here 830bf7b06c88e4adaf03215251995c1069dfd08f and here 5d6b2ecad0f325fc8265c455422fba4155002b08. As of Handel 0.5.7 you can catch errors as follows:

```
Handel.RunHandel(`
start
  *
finish
`).catch((ex) => console.log(ex));
```

As you may have noticed, the RunHandel function now returns a promise. This is because the best practice for Tone.js suggests waiting for Tone.Start() (which is asynchronous) before playing sounds. So that  the Handel Interpreter's errors are not missed, I decided to now make RunHandel an async function.

I hope this satisfies your use case. If not I can reopen the issue.

**Comment by markknol** [comment]
Thanks so much for doing this! Will start playing with it later this week!
